src/openzeppelin/token/erc721/library.cairo

```diff
@@ -70,490 +70,494 @@ end
+            pedersen_ptr: HashBuiltin*,
+            syscall_ptr: felt*,
+            range_check_ptr
+        }(token_id: Uint256):
+        alloc_locals
+        with_attr error_message("ERC721: token_id is not a valid Uint256"):
+            uint256_check(token_id)
+        end
+        let (owner) = owner_of(token_id)
 
-    # Increase receiver balance
-    let (receiver_bal) = ERC721_balances.read(to)
-    let (new_balance: Uint256) = SafeUint256.add(receiver_bal, Uint256(1, 0))
-    ERC721_balances.write(to, new_balance)
+        # Clear approvals
+        _approve(0, token_id)
 
-    # Update token_id owner
-    ERC721_owners.write(token_id, to)
-    Transfer.emit(from_, to, token_id)
-    return ()
-end
+        # Decrease owner balance
+        let (balance: Uint256) = ERC721_balances.read(owner)
+        let (new_balance: Uint256) = SafeUint256.sub_le(balance, Uint256(1, 0))
+        ERC721_balances.write(owner, new_balance)
 
-func _safe_transfer{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }(
-        from_: felt,
-        to: felt,
-        token_id: Uint256,
-        data_len: felt,
-        data: felt*
-    ):
-    _transfer(from_, to, token_id)
-
-    let (success) = _check_onERC721Received(from_, to, token_id, data_len, data)
-    with_attr error_message("ERC721: transfer to non ERC721Receiver implementer"):
-        assert_not_zero(success)
+        # Delete owner
+        ERC721_owners.write(token_id, 0)
+        Transfer.emit(owner, 0, token_id)
+        return ()
     end
-    return ()
-end
 
-func _check_onERC721Received{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }(
-        from_: felt,
-        to: felt,
-        token_id: Uint256,
-        data_len: felt,
-        data: felt*
-    ) -> (success: felt):
-    let (caller) = get_caller_address()
-    let (is_supported) = IERC165.supportsInterface(to, IERC721_RECEIVER_ID)
-    if is_supported == TRUE:
-        let (selector) = IERC721_Receiver.onERC721Received(
-            to,
-            caller,
-            from_,
-            token_id,
-            data_len,
-            data
-        )
+    func _set_token_uri{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(token_id: Uint256, token_uri: felt):
+        uint256_check(token_id)
+        let (exists) = _exists(token_id)
+        with_attr error_message("ERC721_Metadata: set token URI for nonexistent token"):
+            assert exists = TRUE
+        end
 
-        with_attr error_message("ERC721: transfer to non ERC721Receiver implementer"):
-            assert selector = IERC721_RECEIVER_ID
+        ERC721_token_uri.write(token_id, token_uri)
+        return ()
+    end
+
+    func _check_onERC721Received{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(
+            from_: felt,
```

Comment:
Does it make sense to keep this within the namespace? I think this constitutes an "internal" function as defined in the extensibility docs